src/vs/workbench/parts/search/browser/searchActions.ts

```diff
@@ -474,8 +474,10 @@ export abstract class AbstractSearchAndReplaceAction extends Action {
 
```

Comment:
The proper way to do all this action command business is the following:
* Have a command that is self contained and does the actual work
* Use the command everywhere except Action Bar
* For the Action Bar instatiate an Action that will only use the commandService to execute the command

That patter is cleanest and we use it all over the place (especially explorer).
If you do not want to tackle it in this PR then you can create a debt item.

Reply:
I'll create an item to convert search Actions to commands

Reply:
Great